ts/WoltLabSuite/Core/Api.ts

```diff
@@ -0,0 +1,303 @@
+    this._payload = payload;
+
+    return this;
+  }
+
+  disableLoadingIndicator(): this {
+    this._showLoadingIndicator = false;
+
+    return this;
+  }
+
+  async dispatch(): Promise<unknown> {
+    const url = window.WSC_API_URL + "index.php?ajax-proxy/&t=" + Core.getXsrfToken();
+
+    const body: RequestBody = {
+      actionName: this.actionName,
+      className: this.className,
+    };
+    if (this._objectIDs) {
+      body.objectIDs = this._objectIDs;
+    }
+    if (this._payload) {
+      body.parameters = this._payload;
+    }
+
+    const init: RequestInit = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
+        "X-Requested-With": "XMLHttpRequest",
+        "X-XSRF-TOKEN": Core.getXsrfToken(),
+      },
+      body: Core.serialize(body),
+      mode: "same-origin",
+      credentials: "same-origin",
+      cache: "no-store",
+      redirect: "error",
+    };
+
+    if (this._signal) {
+      init.signal = this._signal.signal;
+    }
+
+    // Use a local copy to isolate the behavior in case of changes before
+    // the request handling has completed.
+    const showLoadingIndicator = this._showLoadingIndicator;
+    if (showLoadingIndicator) {
+      AjaxStatus.show();
+    }
+
+    try {
+      const response = await fetch(url, init);
+
+      if (!response.ok) {
+        throw new StatusNotOk(response);
+      }
+
+      const contentType = response.headers.get("content-type");
+      if (!contentType || !contentType.includes("application/json")) {
+        throw new ExpectedJson(response);
+      }
+
+      let json: ResponseData;
+      try {
+        json = await response.json();
+      } catch (e) {
+        throw new InvalidJson(response);
+      }
+
+      if (json.forceBackgroundQueuePerform) {
+        void import("./BackgroundQueue").then((BackgroundQueue) => BackgroundQueue.invoke());
+      }
+
+      return json.returnValues;
+    } catch (error) {
+      if (error instanceof ExpectedJson || error instanceof InvalidJson || error instanceof StatusNotOk) {
+        throw error;
+      } else {
+        // Re-package the error for use in our global "unhandledrejection" handler.
+        throw new ConnectionError(error);
+      }
+    } finally {
+      if (showLoadingIndicator) {
+        AjaxStatus.hide();
+      }
+
+      DomChangeListener.trigger();
+
+      // fix anchor tags generated through WCF::getAnchor()
+      document.querySelectorAll('a[href*="#"]').forEach((link: HTMLAnchorElement) => {
+        let href = link.href;
+        if (href.indexOf("AJAXProxy") !== -1 || href.indexOf("ajax-proxy") !== -1) {
+          href = href.substr(href.indexOf("#"));
+          link.href = document.location.toString().replace(/#.*/, "") + href;
+        }
+      });
+    }
+  }
+}
+
+export class ApiError extends Error {
+  constructor(message: string) {
+    super(message);
+
+    this.name = "ApiError";
+  }
+}
```

Comment:
Same applies for the other classes:

```suggestion
export class ApiError extends Error {
  name = "ApiError"
}
```